File: news-api/src/main/java/com/news/search/controller/WebSearchController.java
package com.news.search.controller;

import com.news.search.controller.request.SearchRequest;
import com.news.search.controller.response.PageResult;
import com.news.search.controller.response.SearchResponse;
import com.news.search.controller.response.StatResponse;
import com.news.search.service.WebApplicationService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RequiredArgsConstructor
@RestController
@Slf4j
@RequestMapping("/api/v1/webs")
public class WebSearchController {
    private final WebApplicationService webApplicationService;

    @GetMapping
    @ResponseBody
    public PageResult<SearchResponse> search(@Valid SearchRequest request){
        return webApplicationService.search(request.getQuery(), request.getPage(), request.getSize());
    }

    @GetMapping("/stats/ranking")
    @ResponseBody
    public List<StatResponse> findTopStats() {
        log.info("[WebSearchController] find top stats");
        return webApplicationService.findTopQuery();
    }
}

File: news-api/src/main/kotlin/com/search/websearch/controller/WebSearchController.kt
package com.search.websearch.controller

import com.search.websearch.controller.request.SearchRequest
import com.search.websearch.controller.response.PageSearchResponse
import com.search.websearch.controller.response.SearchResponse
import com.search.websearch.controller.response.TopRankResponse
import com.search.websearch.service.WebApplicationService
import jakarta.validation.Valid
import org.slf4j.LoggerFactory
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.ResponseBody
import org.springframework.web.bind.annotation.RestController
@RestController
@RequestMapping("/api/v1/webs")
class WebSearchController(
        private val webApplicationService: WebApplicationService
) {

    private val log = LoggerFactory.getLogger(this::class.java)
    @GetMapping
    @ResponseBody
    fun search(@Valid request: SearchRequest): PageSearchResponse<SearchResponse>{
        return webApplicationService.search(request.query, request.page, request.size)
    }

    @GetMapping("/stats/ranking")
    @ResponseBody
    fun findTopStats(): List<TopRankResponse>{
        log.info("webApplicationService.findTopQuery={}", webApplicationService.findTopQuery())
        return webApplicationService.findTopQuery()
    }
}

